crates/iceberg/src/spec/view_metadata.rs

```diff
@@ -0,0 +1,682 @@
+        self.schema_by_id(schema_id)
+            .expect("Current schema id set, but not found in view metadata")
+    }
+
+    /// Returns properties of the view.
+    #[inline]
+    pub fn properties(&self) -> &HashMap<String, String> {
+        &self.properties
+    }
+
+    /// Append view version to view
+    pub fn append_version(&mut self, view_version: ViewVersion) {
+        self.current_version_id = view_version.version_id();
+
+        self.version_log.push(view_version.log());
+        self.versions
+            .insert(view_version.version_id(), Arc::new(view_version));
+    }
+
+    /// Returns view history.
+    #[inline]
+    pub fn history(&self) -> &[ViewVersionLog] {
+        &self.version_log
+    }
+}
+
+/// Manipulating view metadata.
+pub struct ViewMetadataBuilder(ViewMetadata);
+
+impl ViewMetadataBuilder {
+    /// Creates a new view metadata builder from the given view metadata.
+    pub fn new(origin: ViewMetadata) -> Self {
+        Self(origin)
+    }
+
+    /// Creates a new view metadata builder from the given view creation.
+    pub fn from_view_creation(view_creation: ViewCreation) -> Result<Self> {
+        let ViewCreation {
+            location,
+            schema,
+            properties,
+            name: _,
+            representations,
+            default_catalog,
+            default_namespace,
+            summary,
+        } = view_creation;
+        let initial_version_id = super::INITIAL_SEQUENCE_NUMBER;
+        let version = ViewVersion::builder()
+            .with_default_catalog(default_catalog)
+            .with_default_namespace(default_namespace)
+            .with_representations(representations.into_iter().collect::<Vec<_>>())
+            .with_schema_id(schema.schema_id())
+            .with_summary(summary)
+            .with_timestamp_ms(Utc::now().timestamp_millis())
+            .with_version_id(initial_version_id)
+            .build();
+
+        let versions = HashMap::from_iter(vec![(initial_version_id, version.into())]);
+
+        let view_metadata = ViewMetadata {
+            format_version: ViewFormatVersion::V1,
+            view_uuid: Uuid::now_v7(),
+            location,
+            current_version_id: initial_version_id,
+            versions,
+            version_log: Vec::new(),
+            schemas: HashMap::from_iter(vec![(schema.schema_id(), Arc::new(schema))]),
+            properties,
+        };
+
+        Ok(Self(view_metadata))
+    }
+
+    /// Changes uuid of view metadata.
+    pub fn assign_uuid(mut self, uuid: Uuid) -> Result<Self> {
+        self.0.view_uuid = uuid;
+        Ok(self)
+    }
+
+    /// Returns the new view metadata after changes.
+    pub fn build(self) -> Result<ViewMetadata> {
+        Ok(self.0)
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
+#[serde(rename_all = "kebab-case")]
+/// A log of when each snapshot was made.
+pub struct ViewVersionLog {
```

Comment:
in java this is called `ViewHistoryEntry`, maybe worth naming it here similarly?

Reply:
I am also happy with renaming.
I think we should use the same naming we settle on in our discussion above. https://github.com/apache/iceberg-rust/pull/331#discussion_r1675578086